.github/workflows/promote.yaml

```diff
@@ -0,0 +1,34 @@
+name: Promote snap to other tracks and channels
+
+on:
+  workflow_dispatch:
+    inputs:
+      destination-channel:
+        description: 'Destination Channel, e.g. latest/candidate'
+        required: true
+      origin-channel:
+        description: 'Origin Channel, e.g. latest/edge'
+        required: true
+
+jobs:
+  promote-snap:
+    name: Promote snap
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Snapcraft install
+        run: sudo snap install --classic snapcraft
+      - name: Get snap name
+        id: snap
+        run: grep -Po '(?<=^name:)\s*\K\w+' snap/snapcraft.yaml >> $GITHUB_OUTPUT
```

Comment:
What about something simple like grep + awk? Although I can't think of any other edge cases with this approach.

```
$ grep "name:" snap/snapcraft.yaml | awk '{print $2}'
juju-lint
```

Reply:
Hmm, I see your point although using `python3-yaml` like this seems a bit out of place. I would like to see if there's any other ideas as well from others. Otherwise, I'm good to go with your suggestion :smile: 

@aieri @rgildein @jneo8

Reply:
What about `grep -oP '(?<=^name:)\s*\K[a-z|-]*' snap/snapcraft.yaml`

Reply:
@rgildein - this works but it would have the same problem that @samuelallan72 mentioned in his [previous comment](https://github.com/canonical/juju-lint/pull/287#discussion_r1591812247). We could decide to ignore the complexities of parsing yaml (the hacky way) since we're just trying to fetch one key and it's not mission critical. So I'm alright with just grepping the name.

Reply:
oh I was just too restrictive with my regex. Just replace \w+ with \S+ or even .+

Reply:
awk would be totally fine too
```
awk '/name:/ {print $2}' snap/snapcraft.yaml
```